pkg/daemon/daemon.go

```diff
@@ -131,10 +140,36 @@ func (d *Daemon) IsPrefetchDaemon() bool {
 	return d.DaemonMode == config.DaemonModePrefetch
 }
 
+func (d *Daemon) initClient() error {
+	client, err := nydussdk.NewNydusClient(d.APISock())
+	if err != nil {
+		return errors.Wrap(err, "failed to create new nydus client")
+	}
+	d.Client = client
+	return nil
+}
+
+func (d *Daemon) ensureClient(action string) error {
+	var err error
+	if d.Client == nil {
```

Comment:
This check should be put in `sync.Once.Do`, maybe some races occur in concurrency?

Reply:
Done!